src/lib/discord/bot.ts

```diff
@@ -13,13 +17,261 @@ new Command({
 		required: true,
 	} as const)
 	.handle(async i => {
+		const name: string = i.options.name
+		const entry: Entry | null = await fetchEntry(createSlug(name))
+
+		if (!entry) {
+			return {
+				content: `Requested entry "${name}" does not exist, or is unavailable.`,
+				color: 0x650ac9,
+			}
+		}
+
+		let content: string
+
+		if (entry.content.length > 985) {
+			content = `${entry.content.slice(0, 985)}...`
+		} else {
+			content = entry.content
+		}
 		return {
 			embeds: [
 				{
-					title: `Entry "${i.options.name}"`,
+					title: name,
+					url: `${BASE_URL}/entry/${entry.slug}`,
+					description: content,
+					color: 0x650ac9,
 				},
 			],
 		}
 	})
 
+new Command({
+	name: 'random',
+	description: 'View a random Repldex entry',
+}).handle(async data => {
+	const entryCount: number = await countEntries()
+	const entries = await fetchEntries({
+		limit: 1,
+		skip: Math.floor(Math.random() * entryCount),
+	})
+	const entry = entries[Math.floor(Math.random() * entries.length)]
+
+	return {
+		embeds: [
+			{
+				title: entry.title,
+				url: `${process.env.BASE_URL}/entry/${entry.slug}`,
+				description: entry.content,
+				color: 0x650ac9,
+			},
+		],
+	}
+})
+
+// This command has like 2 million issues, so i have temporaily removed it!
+
+new Command({
+	name: 'search',
+	description: 'Search Repldex entries',
+})
+	.addOption({
+		name: 'query',
+		description: 'Query to search for',
+		type: ApplicationCommandOptionType.String,
+		required: true,
+	})
+	.handle(async data => {
+		console.log(data)
+		//query should always be a string. probably a better way to do this
+		if (typeof data.options.query !== 'string') throw new Error('Custom ID is invalid')
+		const query: string = data.options.query
```

Comment:
There is a better way to do this, add `as const` to the object in `addOption`. Sadly TypeScript doesn't let us read the stuff that's inside addOption (name, type, required) without `as const`.

Reply:
will be resolved next commit

Reply:
This issue has been resolved